app/models/episode.rb

```diff
@@ -309,30 +309,28 @@ def set_external_keyword
     return unless !published_at.nil? && keyword_xid.nil?
     identifiers = []
     [:published_at, :guid].each do |attr|
-      # Adzerk does not allow commas or colons in keywords; omitting dashes for space
-      identifiers << self.send(attr)
-                         .to_s
-                         .downcase
-                         .gsub(/[:,-]/,'')
-                         .gsub(/\s+/,'')
-                         .strip
-                         .slice(0, 10)
+      identifiers << sanitize_keyword(self.send(attr), 10)
     end
-    identifiers << (title || 'undefined').to_s
-                                         .downcase
-                                         .gsub(/[:,-]/,' ')
-                                         .gsub(/\s+/,' ')
-                                         .strip
-                                         .slice(0, 20)
+    identifiers << sanitize_keyword(title || 'undefined', 20)
     self.keyword_xid = identifiers.join('_')
   end
 
+  def sanitize_keyword(kw, length)
+    kw.to_s
+      .downcase
+      .gsub(/[^ a-z0-9_-]/,' ')
```

Comment:
In dovetail, @kookster is just blanking these out `''` rather than spaces `' '`.  Do we want to do the same here?